kubernetes-azure-go/main.go

```diff
@@ -0,0 +1,171 @@
+package main
+
+import (
+	"encoding/base64"
+
+	"github.com/pulumi/pulumi-azure-native/sdk/go/azure/containerservice"
+	"github.com/pulumi/pulumi-azure-native/sdk/go/azure/network"
+	"github.com/pulumi/pulumi-azure-native/sdk/go/azure/resources"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi/config"
+)
+
+func main() {
+	pulumi.Run(func(ctx *pulumi.Context) error {
+		// Get some configuration values or set default values
+		cfg := config.New(ctx, "")
+		sshPubKey := cfg.Require("sshPubKey")
+		mgmtGroup := cfg.Require("mgmtGroupId")
+		prefixForDns, err := cfg.Try("prefixForDns")
+		if err != nil {
+			prefixForDns = "pulumi"
+		}
+		kubernetesVersion, err := cfg.Try("kubernetesVersion")
+		if err != nil {
+			kubernetesVersion = "1.24.3"
+		}
+		numWorkerNodes, err := cfg.TryInt("numWorkerNodes")
+		if err != nil {
+			numWorkerNodes = 3
+		}
+		nodeVmSize, err := cfg.Try("nodeVmSize")
+		if err != nil {
+			nodeVmSize = "Standard_DS2_v2"
+		}
+
+		// Create an Azure Resource Group
+		resourceGroup, err := resources.NewResourceGroup(ctx, "resourceGroup")
+		if err != nil {
+			return err
+		}
+
+		// Create an Azure Virtual Network
+		virtualNetwork, err := network.NewVirtualNetwork(ctx, "aks-network", &network.VirtualNetworkArgs{
+			AddressSpace: &network.AddressSpaceArgs{
+				AddressPrefixes: pulumi.StringArray{
+					pulumi.String("10.0.0.0/16"),
+				},
+			},
+			ResourceGroupName: resourceGroup.Name,
+		})
+		if err != nil {
+			return err
+		}
+
+		// Create three subnets in the virtual network
+		subnet1, err := network.NewSubnet(ctx, "subnet-1", &network.SubnetArgs{
+			AddressPrefix:      pulumi.String("10.0.0.0/22"),
+			ResourceGroupName:  resourceGroup.Name,
+			VirtualNetworkName: virtualNetwork.Name,
+		})
+		if err != nil {
+			return err
+		}
+
+		subnet2, err := network.NewSubnet(ctx, "subnet-2", &network.SubnetArgs{
```

Comment:
When I run this, I get compile errors about this being declared but not used. If we don't need to reference `subnet2` elsewhere in the code, we can ignore it with `_`:

```suggestion
		_, err = network.NewSubnet(ctx, "subnet-2", &network.SubnetArgs{
```

Reply:
We definitely need subnet1, but aren't referencing subnet2 (or subnet3) in this template. However, their values might be needed if a user adds more nodepools, so I've added them to the list of exports. Fixed in 8cb24a3.